### DISCUSSION TITLE
Error 905 for tinytuya version 3.4

### DISCUSSION THREAD

**Discussion by onenazlyhelmy** [jasonacox/tinytuya#394] [Q&A]
Hi dev, good day, i hv problem with version 3.4, with response when try to get statu of the device 
set_status() result {'Error': 'Network Error: Device Unreachable', 'Err': '905', 'Payload': None}
by the way i hv information he tuya device after did the scanning :
-------------------------------------
inyTuya (Tuya device scanner) [1.12.9]

Scanning on UDP ports 6666 and 6667 and 7000 for devices for 18 seconds...

Unknown v3.4 Device   Product ID = keyjnuy4s3kre7m7  [Valid Broadcast]:
    Address = 192.168.0.10   Device ID = ebc41c67705ffd80abzqnf (len:22)  Local Key =   Version = 3.4  Type = default, MAC = 
    No Stats for 192.168.0.10: DEVICE KEY required to poll for status
Scan completed in 18.0674 seconds                  
                    
Scan Complete!  Found 1 devices.
Broadcasted: 1
Versions: 3.4: 1

----------------------------------------------------------------
pls need help

**Comment by jasonacox** [comment]
The 3.4 devices will require the "local key". Did you run wizard,  `python -m tinytuya wizard` to get the keys?

**Reply by onenazlyhelmy** [reply]
I ran the wizard, Actually, my point is, i try to get device status locally(LAN) by trigger request on port 6668, the response still the same..i check the source code print out manually, i can  seethe payload is empty, like :
ayload > MessagePayload(cmd=16, payload=b'{}')
IP: 192.168.0.10,  Socket port: 6668
IP: 192.168.0.10,  Socket port: 6668

**Reply by jasonacox** [reply]
You can turn on the debug mode either via command line:

```
python3 -m tinytuya scan -d
```
or via code - something like:
```python
import tinytuya

print("TinyTuya (Tuya Interface) [%s]\n"%tinytuya.__version__)
tinytuya.set_debug(True)

d = tinytuya.OutletDevice(
      dev_id=DEVICEID,
      address=DEVICEIP,
      local_key=DEVICEKEY,
      version=3.4)
print(d)

print(d.detect_available_dps())

print(" > Fetch Status < ")
data = d.status()
print(data)

```

**Reply by onenazlyhelmy** [reply]
I scanned, and result found the devices:

-----------------------------------------------------------------------------------------
TinyTuya (Tuya device scanner) [1.12.9]

Scanning on UDP ports 6666 and 6667 and 7000 for devices for 18 seconds...

Unknown v3.4 Device   Product ID = keyjnuy4s3kre7m7  [Valid Broadcast]:
    Address = 192.168.0.10   Device ID = ebc41c67705ffd80abzqnf (len:22)  Local Key =   Version = 3.4  Type = default, MAC = 
    No Stats for 192.168.0.10: DEVICE KEY required to poll for status
Scan completed in 18.0614 seconds                  
                    
Scan Complete!  Found 1 devices.
Broadcasted: 1
Versions: 3.4: 1
Unknown Devices: 1

>> Saving device snapshot data to snapshot.json
--------------------------------------------------------------
this my simple python script 
------------- -----test1.py -------------------------------
import tinytuya


d = tinytuya.OutletDevice(
    dev_id='ebc41c67705ffd80abzqnf',
    address='192.168.0.10',      # Or set to 'Auto' to auto-discover IP address
    local_key='0123456789abcdef', 
    version=3.4)

tinytuya.set_debug(True)
data = d.status() 

print('Device status: %r' % data)
---------------------------------------- this is simple python script result ------------------------------
root@ubuntu:/storage/development/PYTHONS/tuya/tinytuya# python3 test1.py 
DEBUG:TinyTuya [1.12.9]

DEBUG:Python 3.11.4 (main, Jun  7 2023, 10:13:09) [GCC 12.2.0] on linux
DEBUG:Using PyCrypto (3, 18, '0')
DEBUG:status() entry (dev_type is default)
DEBUG:final payload_dict for 'ebc41c67705ffd80abzqnf' ('v3.4'/'default'): {1: {'command': {'gwId': '', 'devId': '', 'uid': '', 't': ''}}, 7: {'command': {'protocol': 5, 't': 'int', 'data': {}}, 'command_override': 13}, 8: {'command': {'gwId': '', 'devId': ''}}, 9: {'command': {'gwId': '', 'devId': ''}}, 10: {'command': {}, 'command_override': 16}, 13: {'command': {'protocol': 5, 't': 'int', 'data': {}}}, 16: {'command': {}}, 18: {'command': {'dpId': [18, 19, 20]}}, 64: {'command': {'reqType': '', 'data': {}}}}
DEBUG:building command 10 payload=b'{}'
 payload > MessagePayload(cmd=16, payload=b'{}')
IP: 192.168.0.10,  Socket port: 6668
DEBUG:sending payload quick
DEBUG:final payload: b'0123456789abcdef'
DEBUG:payload encrypted=b'000055aa00000001000000030000004472727e881edcfd0100a718687909b565377222e061a924c591cd9c27ea163ed4d7dd1f9eb60b8cb5b90748aa228534c62460d05b84665c9e692efc896dd77d680000aa55'
DEBUG:_recv_all(): no data? b''
DEBUG:_recv_all(): no data? b''
DEBUG:received null payload (None), fetch new one - 1 retries remaining
DEBUG:_recv_all(): no data? b''
DEBUG:_recv_all(): no data? b''
DEBUG:received null payload (None) but out of recv retries, giving up
DEBUG:session key negotiation failed on step 1
IP: 192.168.0.10,  Socket port: 6668

**Reply by jasonacox** [reply]
> local_key='0123456789abcdef',

This is not a valid local key.   If you change it just to keep from posting the real local key, that's fine, just want to make sure you know that 3.4 devices require a valid local key to work at all.

Once you have the real local key, set up a monitoring loop to see if the device is emitting anything:

```python
import time
import tinytuya

d = tinytuya.Device( 'DEVICEID', 'DEVICEIP', local_key='DEVICEKEY', persist=True, version=3.4, connection_timeout=1 )

print( d.status() )

print(" > Begin Monitor Loop <")
pingtime = time.time() + 9
polltime = time.time() + 3

while(True):
    if( pingtime <= time.time() ):
        payload = d.generate_payload(tinytuya.HEART_BEAT)
        data = d.send(payload)
        pingtime = time.time() + 9
        if data:
            print( data )

    # only needed for devices which do not update asynchronously
    #if( polltime <= time.time() ):
    #    polltime = time.time() + 3
    #    print( '================' )
    #    print( d.updatedps([17,18,19,20]) )
    #    print( d.status() )

    data = d.receive()

    if data:
        print( data )

```

**Comment by hamed-jamali-software** [comment]
I have exactly the same problem, did you find a solution?

DEVICEID = "bf5e9d230926e6b12fojsu"
DEVICEIP = "192.168.43.16"
DEVICEKEY = "key8u54q9dtru5jw"
DEVICEVERS = "3.4"





TinyTuya (Tuya device scanner) [1.15.1]

Scanning on UDP ports 6666 and 6667 and 7000 for devices for 50 seconds...

New Broadcast from App at 192.168.43.44 - {'from': 'app', 'ip': '192.168.43.44'}
    Address = 192.168.43.16   Device ID = bf5e9d230926e6b12fojsu (len:22)  Local Key =   Version = 3.4  Type = default, MAC = 
    No Stats for 192.168.43.16: DEVICE KEY required to poll for status
Scan completed in 50.0337 seconds                  
                    
Scan Complete!  Found 1 devices.
Broadcasted: 1
Versions: 3.4: 1
Unknown Devices: 1

>> Saving device snapshot data to snapshot.json

**Reply by uzlonewolf** [reply]
The scanner requires a devices.json file in the working directory.  Make sure that file exists and contains, at a minimum:

```json
[
    {
        "name": "My Device",
        "id": "bf5e9d230926e6b12fojsu",
        "key": "key8u54q9dtru5jw",
        "mac": ""
    }
]
```

**Reply by uzlonewolf** [reply]
Also, what type of device is it?  Bluetooth/Zigbee gateways do not respond to status() requests.

**Reply by hamed-jamali-software** [reply]
Where is this file created?

python -m tinytuya scan -d



The lamp device is Wi-Fi and works well in the original software tuya
Purchased from Amazon today

**Reply by jasonacox** [reply]
Hi @hamed-jamali-software 

The `devices.json` file is created when you run the setup wizard (`python -m tinytuya wizard`) and it is created in the local directory where you run the script.  

A `snapshot.json` file is created when you run the scan as you mention, also in the local directory where you run the script.

**Comment by hemantsaini1301** [comment]
![WhatsApp Image 2025-05-17 at 6 22 54 PM](https://github.com/user-attachments/assets/6a2ab57a-8c15-4d05-865c-66b6db73ba15)


Hi ,

i am facing a issue 


Error : 
{'Error': 'Network Error: Device Unreachable', 'Err': '905', 'Payload': None}


Code : 
```python
import tinytuya
import logging

# Enable debug output for TinyTuya
logging.basicConfig(level=logging.DEBUG)
d = tinytuya.OutletDevice('d772af53ed70790078f7q8', '192.168.31.30', 'ai2cizw27dqsamlr',version=3.4 )

 
 
# Try to get device status
try:
    data = d.status()
    print('Status:', data)
except Exception as e:
    print('Error:', e)

print(" > Fetch Status < ")
data = d.status()
print(data)

print(" > Turn On < ")
d.turn_on()
data = d.status()
print(data)
 

print(" > Turn Off< ")
d.turn_off()
data = d.status()
print(data)
```


can you provide a little bit help for resolved it

**Reply by jasonacox** [reply]
Are you able to ping the device?
